service-workers/service-worker/fetch-event-within-sw-manual.html

```diff
@@ -0,0 +1,150 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/get-host-info.sub.js"></script>
+<script src="resources/test-helpers.sub.js"></script>
+<body>
+<script>
+const worker = 'resources/fetch-event-within-sw-worker.js';
+
+function wait(ms) {
+  return new Promise(r => setTimeout(r, ms));
+}
+
+function reset() {
```

Comment:
Maybe it would be a good idea to not only call reset() at the beginning of your tests, but also have add_completion_callback(reset) somewhere to make sure things get cleaned up after running the tests